app/src/sharedTest/java/org/oppia/android/app/options/AppLanguageFragmentTest.kt

```diff
@@ -133,67 +146,114 @@ class AppLanguageFragmentTest {
       testCoroutineDispatchers.runCurrent()
-      checkSelectedLanguage(ENGLISH)
+      verifyEnglishIsSelected()
     }
   }
 
   @Test
-  fun testAppLanguage_changeLanguageToFrench_selectedLanguageIsFrench() {
-    launch<AppLanguageActivity>(createAppLanguageActivityIntent("English")).use {
-      checkSelectedLanguage(ENGLISH)
-      selectLanguage(FRENCH)
-      checkSelectedLanguage(FRENCH)
+  fun testAppLanguage_changeLanguageToPortuguese_selectedLanguageIsPortuguese() {
+    launch<AppLanguageActivity>(createAppLanguageActivityIntent(OppiaLanguage.ENGLISH)).use {
+      testCoroutineDispatchers.runCurrent()
+      verifyEnglishIsSelected()
+      selectPortuguese()
+      verifyPortugueseIsSelected()
     }
   }
 
   @Test
-  fun testAppLanguage_changeLanguageToFrench_configChange_selectedLanguageIsFrench() {
-    launch<AppLanguageActivity>(createAppLanguageActivityIntent("English")).use {
-      checkSelectedLanguage(ENGLISH)
-      selectLanguage(FRENCH)
+  fun testAppLanguage_changeLanguageToPortuguese_configChange_selectedLanguageIsPortuguese() {
+    launch<AppLanguageActivity>(createAppLanguageActivityIntent(OppiaLanguage.ENGLISH)).use {
+      testCoroutineDispatchers.runCurrent()
+      verifyEnglishIsSelected()
+      selectPortuguese()
       rotateToLandscape()
-      checkSelectedLanguage(FRENCH)
+      verifyPortugueseIsSelected()
     }
   }
 
   @Test
   @Config(qualifiers = "sw600dp")
-  fun testAppLanguage_tabletConfig_changeLanguageToFrench_selectedLanguageIsFrench() {
-    launch<AppLanguageActivity>(createAppLanguageActivityIntent("English")).use {
+  fun testAppLanguage_tabletConfig_changeLanguageToPortuguese_selectedLanguageIsPortuguese() {
+    launch<AppLanguageActivity>(createAppLanguageActivityIntent(OppiaLanguage.ENGLISH)).use {
+      testCoroutineDispatchers.runCurrent()
+      verifyEnglishIsSelected()
+      selectPortuguese()
+      verifyPortugueseIsSelected()
+    }
+  }
+
+  @Test
+  fun testAppLanguage_changeLanguageToSwahili_selectedLanguageObservedIsSwahili() {
+    launch<AppLanguageActivity>(createAppLanguageActivityIntent(OppiaLanguage.ENGLISH)).use {
+      testCoroutineDispatchers.runCurrent()
+      verifyEnglishIsSelected()
+      selectKiswahili()
+      var appLanguageActivity: AppLanguageActivity? = null
+      it.onActivity { it1 -> appLanguageActivity = it1 }
+      testCoroutineDispatchers.runCurrent()
+      appLanguageActivity?.recreate()
       testCoroutineDispatchers.runCurrent()
-      checkSelectedLanguage(ENGLISH)
-      selectLanguage(FRENCH)
-      checkSelectedLanguage(FRENCH)
+      verifyKiswahiliIsSelected(appLanguageActivity)
     }
   }
 
-  private fun checkSelectedLanguage(index: Int) {
+  private fun verifyKiswahiliIsSelected(appLanguageActivity: AppLanguageActivity?) {
+    checkSelectedLanguage(index = KISWAHILI_BUTTON_INDEX, expectedLanguageName = "Kiswahili")
+    assertThat(appLanguageActivity?.appLanguageActivityPresenter?.getLanguageSelected()?.name)
```

Comment:
We shouldn't ever be depending directly on implementations to verify state in tests like this.

Instead, we should verify either via the UI (which is what the user sees) or via the domain controller (which is the true state of the application). Ideally, we should check both between all the tests of the suite for robustness and to ensure UI consistency with domain state.

Reply:
What issues are you running into specifically with verifying using the UI @KevinGitonga?

Reply:
The issue was that the changes were not being reflected on the UI automatically as excpected to be able to verify the same.